.github/workflows/dotcom-components.yml

```diff
@@ -0,0 +1,64 @@
+name: Build dotcom-components
+
+on:
+  pull_request:
+    branches:
+      - "**"
+  push:
+    branches:
+      - main
+
+jobs:
+  dotcom-components:
+    if: >-
+      (github.actor != 'dependabot[bot]') &&
+        (github.event.pull_request.head.repo.owner.login == 'guardian' ||
+          github.event_name == 'push')
+    # Required by actions-assume-aws-role
+    permissions:
+      id-token: write
+      contents: read
+
+    name: dotcom-components
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Use Node.js
+        uses: guardian/actions-setup-node@main
+
+      - name: Test + build
+        run: |
+          npm install -g yarn
+          yarn install
+          yarn setup
+          yarn lint
+          yarn test
+
+          NODE_ENV=production yarn server build
+          NODE_ENV=production NODE_OPTIONS="--max-old-space-size=16384" yarn modules build
```

Comment:
This `NODE_OPTIONS` parameter is only needed for this step?

Reply:
yes the server build manages fine without. Rollup seems to use a lot of memory!